### PR TITLE
feat(x/ecocredit): add params simulations

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #1386 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/marketplace/simulation/operations.go [hunk 1]
```diff
@@ -21,38 +23,47 @@ import (
 
 // Simulation operation weights constants
 const (
-	OpWeightMsgBuy             = "op_weight_msg_buy_direct"        //nolint:gosec
```
Comment (aleem1314) [inline x/ecocredit/marketplace/simulation/operations.go:24]: I'm splitting operations into individual files. operatins.go files becoming too big.

**Comment by aleem1314** [comment]
@ryanchristo Not sure why golangci lint is failing.

File: x/ecocredit/base/simulation/operations.go [hunk 2]
```diff
@@ -244,12 +293,34 @@ func WeightedOperations(
...
 
-	// TODO: #1363
-	// basketOps := basketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, basketQryClient)
-	// ops = append(ops, basketOps...)
-	marketplaceOps := marketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, mktQryClient)
+	basketOps := basketsims.WeightedOperations(appParams, cdc, ak, bk, govk, qryClient, basketQryClient, authority)
+	ops = append(ops, basketOps...)
+	marketplaceOps := marketsims.WeightedOperations(appParams, cdc, ak, bk, qryClient, mktQryClient, govk, authority)
 
 	return append(ops, marketplaceOps...)
```
Comment (ryanchristo) [inline x/ecocredit/base/simulation/operations.go:325]: Ok for now but I'm wondering if we can somehow move this to the module `WeightedOperations` method:

```
// WeightedOperations implements AppModuleSimulation/WeightedOperations.
func (m Module) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
	baseServer, basketServer, marketServer := m.Keeper.QueryServers()
	return basesims.WeightedOperations(
		simState.AppParams,
		simState.Cdc,
		m.accountKeeper,
		m.bankKeeper,
		baseServer,
		basketServer,
		marketServer,
	)
}
```

File: x/ecocredit/basket/simulation/msg_update_basket_fees.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	basetypes "github.com/regen-network/regen-ledger/x/ecocredit/base/types/v1"
+	types "github.com/regen-network/regen-ledger/x/ecocredit/basket/types/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/simulation/utils"
+)
+
+const OpWeightMsgUpdateBasketFees = "op_weight_update_basket_fees" //nolint:gosec
```
Comment (ryanchristo) [inline x/ecocredit/basket/simulation/msg_update_basket_fees.go:20]: Here's where golint is failing. It looks like `msg` is missing in the string and the use of `msg` is what requires the lint ignore comment. Updating the string to include `msg` will make the lint error go away.

**Review by ryanchristo — APPROVED** [review]
ACK. Nice work. I like the idea of splitting up the operations. The golint issue should be resolved by adding `msg` to the op const (see comment below).

File: x/ecocredit/base/simulation/operations.go (hunk 2 above)
Comment (ryanchristo) [reply x/ecocredit/base/simulation/operations.go:325]: Maybe an idea for the future but ideally we wouldn't need to import basket and marketplace operations into the operations for the base package/submodule. Might also be more trouble than its worth.

File: x/ecocredit/basket/simulation/msg_update_basket_fees.go (hunk 3 above)
Comment (ryanchristo) [reply x/ecocredit/basket/simulation/msg_update_basket_fees.go:20]:
```suggestion
const OpWeightMsgUpdateBasketFees = "op_weight_msg_update_basket_fees" //nolint:gosec
```

**Pull request merged**